proto/cosmos/base/reflection/v1beta1/reflection.proto

```diff
@@ -21,6 +25,187 @@ service ReflectionService {
+  // bech32_consensus_address_prefix is the bech32 prefix for consensus nodes addresses
+  string bech32_consensus_address_prefix = 3;
+  // bech32_account_public_key_prefix is the bech32 prefix for accounts public keys
+  string bech32_account_public_key_prefix = 4;
+  // bech32_validator_public_key_prefix is the bech32 prefix for validators public keys
+  string bech32_validator_public_key_prefix = 5;
+  // bech32_consensus_public_key_prefix is the bech32 prefix for consensus nodes public keys
+  string bech32_consensus_public_key_prefix = 6;
+  // purpose is the BIP-0044 purpose code
+  uint32 purpose = 7;
+  // coin_type is the BIP-0044 coin type code
+  uint32 coin_type = 8;
+  // full_fundraiser_path is the BIP-0044 prefix (deprecated in favor of full_bip44_path)
+  string full_fundraiser_path = 9;
+  // full_bip44_path is the BIP-0044 prefix
+  string full_bip44_path = 10;
+}
+
+// MsgDescriptor describes a cosmos-sdk message that can be delivered with a transaction
+message MsgDescriptor {
+  // msg contains a descriptor of sdk.ServiceMsg or sdk.Msg
+  oneof msg {
+    // service_msg is used when the message is an sdk.ServiceMsg type
+    ServiceMsgDescriptor service_msg = 1;
+    // legacy_msg is used when the message is an sdk.Msg type
+    LegacyMsgDescriptor legacy_msg = 2;
+  }
+}
+
+// ServiceMsgDescriptor describes an sdk.ServiceMsg type
+message ServiceMsgDescriptor {
+  // fullname is the protobuf fullname of the given sdk.ServiceMsg
+  string fullname = 1;
+  // route is the sdk.ServiceMsg route, it is equal to type_url
+  string route = 2;
+  // type_url defines the type URL used when marshalling the type as any
+  // this is required so we can provide type safe google.protobuf.Any marshalling and
+  // unmarshalling, making sure that we don't accept just 'any' type
+  // in our interface fields
+  string type_url = 3;
+}
+
+// LegacyMsgDescriptor describes an sdk.Msg type
+message LegacyMsgDescriptor {
+  // fullname is the protobuf fullname of the message
+  string fullname = 1;
+  // type_url defines the type URL used when marshalling the type as any
+  // this is required so we can provide type safe google.protobuf.Any marshalling and
+  // unmarshalling, making sure that we don't accept just 'any' type
+  // in our interface fields
+  string type_url = 2;
+}
+
+// GetAppDescriptorRequest is the request type of the GetAppDescriptor RPC.
+message GetAppDescriptorRequest {}
+
+// GetAppDescriptorResponse is the response type of the GetAppDescriptor RPC.
+message GetAppDescriptorResponse {
+  // app contains the cosmos-sdk application descriptor
+  AppDescriptor app = 1;
+}
+
+// QueryServicesDescriptor contains the list of cosmos-sdk queriable services
+message QueryServicesDescriptor {
+  // query_services is a list of cosmos-sdk QueryServiceDescriptor
+  repeated QueryServiceDescriptor query_services = 1;
+}
+
+// QueryServiceDescriptor describes a cosmos-sdk queryable service
+message QueryServiceDescriptor {
```

Comment:
This covers only the gRPC queriers, it only adds one more thing: the tendermint query path, which can be used if a client desires to query via tendermint rather than gRPC. Plus it holds a space in case in the future we desire to add more information on specific query services. For example: module name would be useful to identify to which module that query method belongs to, without needing to parse the servicedescriptor fullname

Reply:
> it only adds one more thing: the tendermint query path, which can be used if a client desires to query via tendermint rather than gRPC.

But the tendermint query path (`full_query_path` below) is exactly the gRPC method fullname.

I would remove `repeated QueryMethodDescriptor methods = 2`, we just need `string fullname = 1` and you can further reflection to list all methods.